src/workers/downloadWorker.test.ts

```diff
@@ -0,0 +1,271 @@
+      expect(mockCallbacks.onSuccess).toHaveBeenCalledWith('123');
+    });
+
+    it('When an error occurs, then the error callback is called', async () => {
+      const error = new Error('Download failed');
+      (createFileDownloadStream as any).mockRejectedValue(error);
+
+      await downloadingFile(mockParams, mockCallbacks);
+
+      expect(mockCallbacks.onError).toHaveBeenCalledWith({});
+      expect(mockCallbacks.onSuccess).not.toHaveBeenCalled();
+    });
+
+    test('When the download progress changes, then the callback is called', async () => {
+      const mockedProgressValue = 75;
+      let progressCallback: any;
+
+      (createFileDownloadStream as any).mockImplementation((file, workspace, callback) => {
+        progressCallback = callback;
+        return Promise.resolve({
+          getReader: () => ({
+            read: vi
+              .fn()
+              .mockResolvedValueOnce({ done: false, value: new Uint8Array([1, 2, 3]) })
+              .mockResolvedValueOnce({ done: true, value: undefined }),
+            releaseLock: vi.fn(),
+          }),
+        });
+      });
+
+      await downloadingFile(mockParams, mockCallbacks);
+      progressCallback(mockedProgressValue);
+
+      expect(mockCallbacks.onProgress).toHaveBeenCalledWith(mockedProgressValue);
+    });
+
+    test('When the abort controller is provided, then it is used if needed', async () => {
+      const customAbortController = new AbortController();
+
+      const mockReader = {
+        read: vi
+          .fn()
+          .mockResolvedValueOnce({ done: false, value: new Uint8Array([1, 2, 3]) })
+          .mockResolvedValueOnce({ done: true, value: undefined }),
+        releaseLock: vi.fn(),
+      };
+
+      const mockStream = {
+        getReader: vi.fn().mockReturnValue(mockReader),
+      };
+
+      (createFileDownloadStream as any).mockResolvedValue(mockStream);
+
+      await downloadingFile(mockParams, mockCallbacks, customAbortController);
+
+      expect(createFileDownloadStream).toHaveBeenCalledWith(
+        mockFile,
+        false,
+        mockCallbacks.onProgress,
+        customAbortController,
+        mockParams.credentials,
+      );
+    });
+  });
+
+  describe('Downloading the file as Blob', () => {
+    test('When the download starts, then should create a blob from the stream chunks', async () => {
+      const mockReader = {
+        read: vi
+          .fn()
+          .mockResolvedValueOnce({ done: false, value: new Uint8Array([1, 2, 3]) })
+          .mockResolvedValueOnce({ done: false, value: new Uint8Array([4, 5, 6]) })
+          .mockResolvedValueOnce({ done: true, value: undefined }),
+        releaseLock: vi.fn(),
+      };
+
+      const mockStream = {
+        getReader: vi.fn().mockReturnValue(mockReader),
+      };
+
+      const onBlobReady = vi.fn();
+
+      await downloadUsingBlob(mockStream as any, onBlobReady);
+
+      expect(onBlobReady).toHaveBeenCalledWith(expect.any(Blob));
```

Comment:
Ideally, you should match the blob to ensure it is the same content, in order to check that no corruption happens during the parts merging